### PR TITLE
Exporter: add support for MIR

### PR DESCRIPTION
**Pull request by W95Psp**
Hax's exporter was previously only focusing on THIR.

The Rust compiler has different representations:
 - Surface AST;
 - HIR: the High-Level Intermediate Representation before being typing;
 - THIR: same, but after typing;
 - MIR: after borrow-checking.
 
THIR and MIR however describe only bodies (i.e. the body of a function, of a constant... anything that is actual code), and are wrapped into HIR items.

This PR makes items be generic in their body. One can thus ask for an `Item<X>`, with `X` being a body type.
This PR also adds a reflection of the MIR AST. Thus, available body types are `ThirBody` and `MirBody<K>` (where `K` is either `Built` or `Const` --for now). Note `(X, Y)` is also a body when `X` and `Y` are bodies.

Finally, this PR changes the `json` subcommand in the CLI:
```
Export directly as a JSON file

Usage: hax json [OPTIONS]

Options:
  -o, --output-file <OUTPUT_FILE>  Path to the output JSON file, "-" denotes stdout [default: hax_frontend_export.json]
  -k, --kind <KIND>                Wether the bodies are exported as THIR, built MIR, const MIR, or a combination. Repeat this option to extract a combination (e.g. [-k thir -k mir-built]) [default: thir] [possible values: thir, mir-built, mir-const]
  -h, --help                       Print help
```

This is part of @sonmarcho and I effort to unify some part of Charon with Hax's exporter.

### CODE REVIEW

**Review by franziskuskiefer — APPROVED** [review]
Yay 🎉
Only a few nits.

File: frontend/exporter/adt-into/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![feature(trait_alias)]
```
[the remark this answers is not in the document]
Comment (W95Psp) [reply frontend/exporter/adt-into/src/lib.rs]: You're right, I recall using a trait alias at some point, but I guess I remove it

File: cli/driver/Cargo.toml [hunk 2]
```diff
@@ -23,6 +23,7 @@ hax-diagnostics.workspace = true
 which.workspace = true
 inquire = "0.6"
 const_format = "0.2"
+itertools.workspace = true
```
[the remark this answers is not in the document]
Comment (W95Psp) [reply cli/driver/Cargo.toml]: You're right, it's actually useless!

File: .utils/rebuild.sh
[the remark this answers is not in the document]
Comment (W95Psp) [reply .utils/rebuild.sh:1]: For me that works, that's what I use to dev

**Pull request merged**